clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -324,8 +342,10 @@ SYCLToolChain::TranslateArgs(const llvm::opt::DerivedArgList &Args,
 Tool *SYCLToolChain::buildBackendCompiler() const {
   if (getTriple().getSubArch() == llvm::Triple::SPIRSubArch_fpga)
     return new tools::SYCL::fpga::BackendCompiler(*this);
-  // fall through is GEN
-  return new tools::SYCL::gen::BackendCompiler(*this);
+  else if (getTriple().getSubArch() == llvm::Triple::SPIRSubArch_gen)
```

Comment:
```suggestion
  if (getTriple().getSubArch() == llvm::Triple::SPIRSubArch_gen)
```
`else` is not needed.